astropy/units/equivalencies.py

```diff
@@ -508,6 +508,59 @@ def convert_K_to_Jy(x_K):
     return [(astrophys.Jy, si.K, convert_Jy_to_K, convert_K_to_Jy)]
 
 
+def thermodynamic_temperature(disp, T_cmb=2.7255*si.K):
+    r"""Defines the conversion between Jy/beam and "thermodynamic temperature",
+    :math:`T_{CMB}`, in Kelvins.  The thermodynamic temperature is a unit very
+    commonly used in cosmology.  See, e.g., "Planck 2013 results. IX. HFI spectral response"
+    (Planck Collaboration 2013) eqn 8 (available on `arXiv <https://arxiv.org/abs/1303.5070>`__).
+
+    :math:`T_{CMB} \equiv B_\nu / \left(2 k \nu^2 / c^2  f(\nu) \right)`
+
+    with :math:`f(\nu) = \frac{ x^2 e^x}{(e^x - 1 )^2}` where :math:`x = h \nu / k T`
+
+    Parameters
+    ----------
+    disp : `~astropy.units.Quantity` with spectral units
+        The observed `spectral` equivalent `~astropy.units.Unit` (e.g.,
+        frequency or wavelength)
+    T_cmb :  `~astropy.units.Quantity` with temperature units (default to 2.7255 K)
+        The CMB temperature at z=0
+
+    Notes
+    -----
+    For broad band receivers, this conversion do not hold
+    as it highly depends on the frequency
+
+
+    Examples
+    --------
+    Planck HFI 143 GHz::
+
+        >>> import numpy as np
+        >>> from astropy import units as u
+        >>> freq = 143*u.GHz
+        >>> equiv = u.thermodynamic_temperature(freq)
+        >>> u.K.to(u.MJy/u.sr, equivalencies=equiv)  # doctest: +FLOAT_CMP
+        379.93172115555564
+
+    """
+    nu = disp.to(si.GHz, spectral())
+
+    def f(nu, T_cmb=T_cmb):
+        x = _si.h * nu / _si.k_B / T_cmb
+        return x**2 * np.exp(x) / (np.exp(x) - 1)**2
+
```

Comment:
Could we use [`np.expm1(x)`](https://docs.scipy.org/doc/numpy/reference/generated/numpy.expm1.html) for better accuracy for small values of x?